handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -399,32 +492,33 @@ public void run() {
         }
     }
 
-    private final class WriterIdleTimeoutTask implements Runnable {
-
-        private final ChannelHandlerContext ctx;
+    private final class WriterIdleTimeoutTask extends AbstractIdleTask {
 
         WriterIdleTimeoutTask(ChannelHandlerContext ctx) {
-            this.ctx = ctx;
+            super(ctx);
         }
 
         @Override
-        public void run() {
-            if (!ctx.channel().isOpen()) {
-                return;
-            }
+        protected void run(ChannelHandlerContext ctx) {
 
             long lastWriteTime = IdleStateHandler.this.lastWriteTime;
             long nextDelay = writerIdleTimeNanos - (System.nanoTime() - lastWriteTime);
             if (nextDelay <= 0) {
                 // Writer is idle - set a new timeout and notify the callback.
                 writerIdleTimeout = ctx.executor().schedule(
                         this, writerIdleTimeNanos, TimeUnit.NANOSECONDS);
+
+                boolean first = firstWriterIdleEvent;
+                if (firstWriterIdleEvent) {
```

Comment:
consider just setting the variable unconditionally. conditionals can be expensive and this variable is not volatile.